package.json

```diff
@@ -141,6 +141,9 @@
   "xo": {
     "ignore": [
       "cli.js"
-    ]
+    ],
+		"rules": {
+			"no-loop-func": 1
```

Comment:
This will avoid that error (and be faster):
```js
var i = fromIndex;
function runAsync(ret) {
  self._addResult(ret);
  return self._run(i + 1);
}
for(; i < length; i++) {
  // ...
  if (isPromise(result)) {
    return result.then(runAsync);
  }
}
```